src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -95,28 +95,31 @@ public static IServiceCollection AddAsyncInitializer(this IServiceCollection ser
         /// <param name="services">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> to add the service to.</param>
         /// <param name="initializer">The delegate that performs async initialization.</param>
         /// <returns>A reference to this instance after the operation has completed.</returns>
-        public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Func<Task> initializer)
+        public static IServiceCollection AddAsyncInitializer(this IServiceCollection services, Func<CancellationToken, Task> initializer)
         {
-            if (initializer == null)
-                throw new ArgumentNullException(nameof(initializer));
-
-            return services
+            return initializer == null
+                ? throw new ArgumentNullException(nameof(initializer))
+                : services
```

Comment:
Please undo this change, it makes the code less readable.
(and it's not related to the feature you're implementing anyway)